### PR TITLE
Remove Sorbet

### PR DESCRIPTION
**Pull request by jasonroelofs**
## Description
While we at WorkOS do prefer strongly typed code where we can, the use of Sorbet in this library is starting to prove more of a hindrance than a help. Specifically, supporting libraries we use like `rbi` and `tapioca` now no longer support Ruby 2.7, making our own support of older versions of Ruby difficult to accomplish.

We also aren't really sure how much of an actual benefit Sorbet provides for public libraries like this one, and we have had requests to drop extra dependencies like `sorbet-runtime`.

This change will enable us to have a much smaller dependency set, allow others to more easily contribute to the library, and allow us to properly test and support both older and newer versions of Ruby.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/workos/user_management.rb [hunk 1]
```diff
@@ -465,15 +378,6 @@ def authenticate_with_magic_auth(
...
-      sig do
-        params(
-          client_id: String,
-          organization_id: String,
-          pending_authentication_token: String,
-          ip_address: T.nilable(String),
-          user_agent: T.nilable(String),
-        ).returns(WorkOS::AuthenticationResponse)
-      end
       def authenticate_with_organization_selection(
```
[the remark this answers is not in the document]
Comment (jasonroelofs) [reply lib/workos/user_management.rb:381]: As in, between the method name and the comment?

[the remark this answers is not in the document]
Comment (alisherry) [reply lib/workos/user_management.rb:381]: Ah it looked like the `end` from the previous method was touching the top of this methods comment, nvm 😄

**Comment by alisherry** [comment]
We should remove `typed` comments from here:
https://github.com/workos/workos-ruby/blob/main/lib/workos.rb#L2
https://github.com/workos/workos-ruby/blob/main/spec/spec_helper.rb#L2
https://github.com/workos/workos-ruby/blob/327fa9b4bc56473031d703602eae5115c7dc863c/lib/workos/version.rb#L2
https://github.com/workos/workos-ruby/blob/327fa9b4bc56473031d703602eae5115c7dc863c/lib/workos/verify_challenge.rb#L2
https://github.com/workos/workos-ruby/blob/main/spec/support/shared_examples/client_spec.rb#L2

**Review by alisherry — APPROVED** [review]
LGTM outside of the comment removals

**Comment by jasonroelofs** [comment]
Ah, yes I did miss some of those, one sec.

**Pull request merged**